Project/Shop.cs

```diff
@@ -43,28 +46,58 @@ void Update()
             //set to be 6-10 seconds, should also be affected by the managers and area in future dev
             randomUpdateTime = Random.Range(6, 10);
         }
-        
+
     }
 
     // return the money to update. Now the amount is based only on costs and level. Should also be affected by the managers.
     void UpdateMoney()
     {
-        player.money += costs + costs / -1000 * Random.Range(-.1f, level);
+        float managerIncome = 0.0;
+        float managerCost = 0.0;
+        for (int i = 0; i < managers.Length; i ++)
+        {
+            managerIncome += managers[i].getLevel() * managers[i].initialIncome * 100;
+            managerCost += managers[i].getLevel() * managers[i].cost * 50;
+        }
+        income += managerIncome;
+        costs += managerCost; 
+        profits += income - costs;
     }
 
     public Shop()

```

Comment:
Update shop constructor to take in Gameclass object so that you can set the player variable

Reply:
You need a shop constructor that takes in a `GameClass` parameter in order for the constructor call on line 62 of `GameClass.cs` to work. Make sure to set `player` to the `GameClass` that's passed in to the constructor